var/spack/repos/builtin/packages/neuron/package.py

```diff
@@ -368,7 +369,7 @@ def archdir(self):
         specific directory by looking for a specific binary.
         """
         if self.spec.satisfies("+cmake"):
-            neuron_arch = str(self.spec.architecture).split("-")[2]
+            neuron_arch = subprocess.check_output(['uname', '-p']).decode('UTF-8').rstrip()
```

Comment:
I wonder if there is python way but...

Reply:
There is the function `os.uname()` but it reports the `machine architecture` and not the `processor` architecture which could (?) be different (?)

Reply:
not sure...but I wouldn't spend too much time if its not easy to verify...

Reply:
I believe that for modern systems those should be the same. In short and according to man page:
```
   -m, --machine
          print the machine hardware name

   -p, --processor
          print the processor type or "unknown"
```
For modern x86 systems I believe that the machine and processor architecture are the same (`x86_64`). I don't know if something changes for `powerpc` or `arm` architectures and less typical systems like `Apple Silicon`, `Raspberry PIs` or future heterogeneous architectures